src/anemoi/inference/inputs/cutout.py

```diff
@@ -0,0 +1,72 @@
+# (C) Copyright 2025 Anemoi contributors.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+
+import logging
+
+import numpy as np
+
+from ..input import Input
+from . import create_input
+from . import input_registry
+
+LOG = logging.getLogger(__name__)
+
+
+@input_registry.register("cutout")
+class Cutout(Input):
+    """Combines one or more LAMs into a global source using cutouts."""
+
+    def __init__(self, context, **sources):
+        """Create a cutout input from a list of sources.
+
+        Parameters
+        ----------
+        context : dict
+            The context runner.
+        sources : dict of sources
+            A dictionary of sources to combine.
+        """
+        super().__init__(context)
+
+        self.sources: dict[str, Input] = {}
+        for src, cfg in sources.items():
+            self.sources[src] = create_input(context, cfg)
+
+    def __repr__(self):
+        return f"Cutout({self.sources})"
+
+    def create_input_state(self, *, date=None):
+        """Create the input state dictionary."""
+
+        LOG.info(f"Concatenating states from {self.sources}")
+        src = list(self.sources.keys())
+        state = self.sources[src[0]].create_input_state(date=date)
+        for i, source in enumerate(src[1:]):
+            _state = self.sources[source].create_input_state(date=date)
+
+            # NOTE: we must decide whether these come from the checkpoint or the input
```

Comment:
Do we need to decide that here? The underlying ekdinput should take care of that - using lat/lon from the checkpoint if they exist, otherwise try to use them from the input.

Reply:
Yes, sorry, that's what I meant. I just wasn't sure about touching that part, we might leave it for another PR but I left a note. For the time being the checkpoint is still used.